### PR TITLE
Port DocsApiUtils and tests

### PR DESCRIPTION
**Pull request by EricBorczuk**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/util/DocsApiUtils.java [hunk 1]
```diff
@@ -263,4 +384,36 @@ public static boolean isRowOnPath(
...
+    Map<String, Object> bindMap = new LinkedHashMap<>(maxDepth + 5);
+
+    bindMap.put("key", null);
+
+    for (int i = 0; i < maxDepth; i++) {
+      String value = "";
+      if (i < path.size()) {
+        value = path.get(i);
+      }
+      bindMap.put("p" + i, value);
```
Comment (ivansenic) [inline sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/util/DocsApiUtils.java:409]: This should be `properties.tableProperties().pathColumnName(i)`..

**Review by ivansenic — APPROVED** [review]
Check comments and merge freely once done..

File: sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/util/DocsApiUtils.java [hunk 2]
```diff
@@ -118,6 +143,102 @@ public static String leftPadTo6(String value) {
...
+        || replaced.contains("'")
+        || replaced.contains("\\");
+  }
+
+  /**
+   * Converts a JSON path string (e.g. "$.a.b.c[0]") into a JSON path string that only uses square
+   * brackets to denote pathing (e.g. "$['a']['b']['c'][0]". This is to allow escaping of certain
+   * characters, such as space, $, and @.
+   */
+  public static String convertJsonToBracketedPath(String path) {
```
[the remark this answers is not in the document]
Comment (EricBorczuk) [reply sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/util/DocsApiUtils.java:202]: It seems to be used in DocsShredder#convertJsonNodeToBoundVariables

[the remark this answers is not in the document]
Comment (ivansenic) [reply sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/util/DocsApiUtils.java:208]: But we are not using the `DocShredder` anymore or? Can you remove in the followup PR if so.. Feel free to have a PR for removing `DocShredder` and this also on the V1?

**Pull request merged**